server/routes/ticket.ts

```diff
@@ -0,0 +1,39 @@
+import { Router } from 'express';
+import { TicketModel } from "../models/tickets/ticket.model";
+
+const router = Router();
+
+router.post('/', async (req, res, next) => {
+    try {
+        const createTicket = await TicketModel.create(req.body);
+        return res.status(201).json(createTicket)
+    }catch (e) {
+        res.status(400).json(e)
+    }
+
+});
+router.put('/:id', async (req, res, next) => {
+    try {
+        const updatedTicket = await TicketModel.findByIdAndUpdate(req.params.id, req.body);
+        res.status(200).json(updatedTicket)
+    } catch (e) {
+        res.status(400).json(e)
+    }
```

Comment:
Forgot to comment on this. Trying to do an operation and catching all errors isn't a good programming pattern. You should prefer validating data and returning meaningful errors rather than catching everything a returning an internal error object that could possibly contain confidential information.

In this case, I would do something like this (note that I am coding this from the top of my head right now, so it might not work if copy-pasted directly):
```ts
router.put('/:id', async (req, res, next) => {
    const id = req.params.id;
    const ticket: ITicket = req.body;
    
    if (!id || !ticket) { 
        return res.status(400).json({"error": "Missing ticket id and/or ticket body." });    
    }
    
    const updatedTicket = await TicketModel.findByIdAndUpdate(id, ticket);
    if (!updatedTicket) { 
        return res.status(404).json({"error": "Ticket with this id doesn't exist.", "id": id});
    }
    return res.status(200).json(updatedTicket); 
});
```

Reply:
Also, prefer using StatusCodes.OK, StatusCodes.BadRequest, etc. as they are more descriptive than raw status codes.